content/src/debug.js

```diff
@@ -201,38 +360,199 @@ function reportSeeeval(method, debugId, length, coordId, elem, args){
+        currentRecordID = debugId;
+        record.startCoords[coordId] = collectCoords(elem);
+        if (debugMode){  
+          showVariables(debugId);
+        }
+      }
     }
   }
 }
 
 function reportResolve(method, debugId, length, coordId, elem, args){
-  var recordD = debugRecordsByDebugId[debugId];
-  if (recordD) {
-    if (!recordD.seeeval) {
-      var recordL = debugRecordsByLineNo[recordD.line];
-      recordD.method = method;
-      recordL.method = method;
-      var index = recordD.eventIndex;
-      var location = traceEvents[index].location.first_line
-      recordD.endCoords[coordId] = collectCoords(elem);
-      recordL.endCoords[coordId] = collectCoords(elem);
-      untraceLine(location);
+  
+  var record = debugRecordsByDebugId[debugId];
+  if (record) {
+    if (!record.seeeval){
+      view.arrow(view.paneid('left'), arrows, -1, false);
+      record.method = method;
+      record.animated = true;
+      var index = record.eventIndex;
+      var line = traceEvents[index].location.first_line
+      if (index > 0){
+        var prevLine = traceEvents[index -1].location.first_line;
+      } else{
+        var prevLine = -1;
+      }
+      record.endCoords[coordId] = collectCoords(elem);
+      untraceLine(index);
+    }          
+  }
+}
+
+function end_program(){
+  //goes back and traces unanimated lines at the end of programs.
+  var currentLine = -1; 
+  var tracedIndex = -1;
+  var justEnded = (currentRecordID <= currentDebugId);
+  var prevLine =  -1;
+  if (traceEvents[prevIndex]){
+    prevLine = traceEvents[prevIndex].location.first_line; 
+  } 
+  while (currentRecordID <= currentDebugId){
+
+    var currentRecord = debugRecordsByDebugId[currentRecordID];
+    var currentIndex = currentRecord.eventIndex;
+    var currentLocation = traceEvents[currentIndex].location;
+    currentLine = currentLocation.first_line;
+
+   if (prevIndex != -1) {
+      var prevLocation = traceEvents[prevIndex].location;
+      var prevLine = prevLocation.first_line;
+    } else{
+      var prevLocation = null;
+      var prevLine = -1;
+    }
+    
+    if (tracedIndex != -1){
+        untraceLine(tracedIndex);
+        tracedIndex = -1;
+    }
+    if(currentLine < prevLine && currentIndex == prevIndex + 1){
+          
+        if (arrows[prevIndex] != null){
+          arrows[prevIndex]['after'] =  {first: currentLocation, second: prevLocation};
+        } else{
+          arrows[prevIndex] = {before: null, after: {first: currentLocation, second: prevLocation}};
+        }
+        if (arrows[currentIndex] != null){
+          arrows[currentIndex]['before'] =  {first: currentLocation, second: prevLocation};
+        } else{
+          arrows[currentIndex] = {before: {first: currentLocation, second: prevLocation}, after : null};
+        }
+        if (!arrows[currentRecordID]){
+          debugRecordsByLineNo[currentLine] = currentRecordID;
+        }
+        //Draw an arrow if the debugger is on
+        if (debugMode) {
```

Comment:
If debug mode is off, do you want to do all the above maintenance of the arrows[] state? It's okay if you do -- maybe add a comment about why you do. Can you turn debug mode on and off during a run? Or something like that?